sql/core/src/test/scala/org/apache/spark/sql/execution/QueryExecutionSuite.scala

```diff
@@ -336,6 +337,18 @@ class QueryExecutionSuite extends SharedSparkSession {
     }
   }
 
+  test("SPARK-47033: EXECUTE IMMEDIATE USING does not recognize session variable names") {
```

Comment:
just realized that this test is in `QueryExecutionSuite`. This is a bit weird. Given that we do not have an end-to-end scala test sutie for EXECUTE IMMEDIATE, shall we add a new `ExecuteImmediateEndToEndSuite`?

Reply:
Agreed, on it.